doc/examples/perception_pipeline/perception_pipeline_tutorial.rst

```diff
@@ -1,198 +1,99 @@
 
- roslaunch moveit_tutorials obstacle_avoidance_demo.launch
+.. image:: images/perception_pipeline_demo_gazebo_screen.png
 
-You should see something like the image shown at the beginning of this tutorial.
-If not, you may have run into a `known OpenGL rendering issue <http://wiki.ros.org/rviz/Troubleshooting>`_. To work around the issue, you can force CPU-based rendering with this command:
+.. image:: images/perception_pipeline_depth_camera_environment.png
 
- export LIBGL_ALWAYS_SOFTWARE=1
+For more explanation of ``depth_camera_environment.launch.py``, you can look at comments in :codedir:`depth_camera_environment.launch.py <examples/perception_pipeline/launch/depth_camera_environment.launch.py>` on GitHub.
 
-You can test obstacle avoidance for yourself by setting the goal state manually and then planning and executing. To learn how to do that look at :doc:`MoveIt Quickstart in RViz </doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial>`
+.. tutorial-formatter:: ./launch/depth_camera_environment.launch.py
 
-Detecting and Adding Object as Collision Object
------------------------------------------------
+For the commands in Shell 2, we must save both camera topics and tf topics due to the fact that MoveIt perception pipeline needs to listen TF's in order to convert the coordinates of pointcloud points according to ``world`` frame. Moreover, the reason of publishing static tf from world to camera frames in ``depth_camera_environment.launch.py`` is that it's necessary to determine transformation between robot and poincloud and that MoveIt's sensor plugins uses this transformation later.
 
-In this section, we will demonstrate an example of extracting a cylinder from a pointcloud, computing relevant values and adding it as a collision object to the planning scene.
-We will be working with point clouds but it can be implemented similarly with depth maps.
+By the way, you can also use :codedir:`this rviz file <examples/perception_pipeline/rviz2/depth_camera_environment.rviz>` on Github to visualize poincloud topics in rviz.
 
-After running the code, you should be able to see something like this in rviz:
+In next step, it will be used this recorded bag file for octomap creation.
 
-.. image:: cylinder_collision_object.png
-   :width: 700px
 
-Running the Code
-++++++++++++++++
-Roslaunch the launch file to run the code directly from moveit_tutorials: ::
+Configuration For 3D Sensors
+----------------------------
+Now whatever you get bag file, you will see ``/camera_1/points``, ``/camera_2/points``, ``/tf`` and ``/tf_static`` in both way when played bag file. It should be created following config file for MoveIt to process these pointcloud topics in planning pipeline. You can also go to :codedir:`here <examples/perception_pipeline/config/sensors_3d.yaml>` for seeing all ``sensors_3d.yaml`` config file on Github.
```

Comment:
Do you mind rewording this to be a bit more precise too? I am not sure if I understand it correctly.

Reply:
I reworded it here and i added another part based on your suggestion.

https://github.com/CihatAltiparmak/moveit2_tutorials/blob/bc1504d3d6c13e17c19bf506fe70358bbc83a100/doc/examples/perception_pipeline/perception_pipeline_tutorial.rst?plain=1#L57-L62